src/_utils/search/create-index/webpack-plugin.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+import { RawSource } from "webpack-sources"
+import PhenomicLoaderWebpackPlugin from "../../../loader/plugin.js"
+import createIndex from "./index"
+
+function CreateSearchIndexWebpackPlugin(options: Object) {
+  this.options = options
+}
+
+CreateSearchIndexWebpackPlugin.prototype.apply = function(compiler) {
+  compiler.plugin("compilation", (compilation/* , params */) => {
+    compilation.plugin("additional-assets", (callback) => {
+      if (!PhenomicLoaderWebpackPlugin.collection) {
+        throw new Error(
+          "Missing Phenomic collection in webpack compilation object. " +
+          "This probably means you are playing with fire."
+        )
+      }
+      const collection = PhenomicLoaderWebpackPlugin.collection
```

Comment:
Yeah (currently), I hope to offer a better API soon. That's what @thangngoc89 pointed out here https://github.com/MoOx/phenomic/issues/621#issuecomment-247192443